### PR TITLE
Feature/adding

### PR DESCRIPTION
**Pull request by Olga-Mishareva**

### CODE REVIEW

File: components/ItemForm.js [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+      <Label htmlFor="description">Description:</Label>
+      <Textarea
+        type="text"
+        name="description"
+        id="description"
+        rows="7"
+        required
+      />
+      <ButtonsWrapper>
+        <BackLink href={"/"} aria-label="cancel">
```
Comment (NikRoe) [inline components/ItemForm.js:35]: your link should also work without the brackets

```suggestion
        <BackLink href="/" aria-label="cancel">
```

File: pages/create.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import useSWR from "swr";
+import { useRouter } from "next/router";
+
+import ItemForm from "@/components/ItemForm";
+
+export default function CreatePage() {
+  const router = useRouter();
+
+  async function addItem(data) {
+    await fetch("api/items", {
```
Comment (NikRoe) [inline pages/create.js:10]: you should wrap your fetch in a `try/catch`-block

File: pages/index.js [hunk 3]
```diff
@@ -12,21 +12,29 @@ export default function HomePage() {
...
-          <h2>Loading...</h2>
-        </li>
-      ) : (
-        items.map((item) => (
-          <li key={item.itemId}>
-            <StyledLink href={`/items/${item.itemId}`}>
-              <Item title={item.title} initialStatus={item.initiallyLost} />
-            </StyledLink>
+    <>
+      <AddItemLink href={"/create"}>
```
Comment (NikRoe) [inline pages/index.js:16]: this link should work fine without the brackets

```suggestion
      <AddItemLink href="/create">
```

File: pages/api/items/index.js [hunk 4]
```diff
@@ -1,16 +1,31 @@
...
+      if (!newItem) {
+        response.status(400).json({ message: "Bad request" });
+        return;
+      }
+
+      response.status(200).json(newItem);
+      break;
+    }
     default: {
       res
```
Comment (NikRoe) [inline pages/api/items/index.js:26]:
```suggestion
      response
```

File: pages/index.js [hunk 5]
```diff
@@ -12,21 +12,29 @@ export default function HomePage() {
...
+      <StyledList>
+        {isLoading ? (
+          <li>
+            <h2>Loading...</h2>
           </li>
-        ))
-      )}
-    </StyledList>
+        ) : (
+          items
```
Comment (NikRoe) [inline pages/index.js:25]: this is a fine solution with method chaining, however, you could also choose an approach where most of the logic is not handled within the jsx.
You could for instance have a variable called `reversedItems` that stores a reversed copy of your items and within your jsx you only map over the `reversedItems`.

**Review by NikRoe — CHANGES_REQUESTED** [review]
Hi Olga,

very nice!

I left you some notes and a few remarks. Please take a look at those :)

File: pages/index.js (hunk 5 above)
Comment (Olga-Mishareva) [reply pages/index.js:25]: It's temporary. I'll add a map next week, so I don't need reverse() function anymore.

File: components/ItemForm.js [hunk 6]
```diff
@@ -0,0 +1,133 @@
+import styled from "styled-components";
+import Link from "next/link";
+
+export default function ItemForm({ onSubmit }) {
+  function handleSubmit(event) {
```
[the remark this answers is not in the document]
Comment (Olga-Mishareva) [reply components/ItemForm.js:5]: User will be redirected to the home page after submission. But I added autoFocus to select. It's more visible now  :)

File: pages/api/items/index.js [hunk 7]
```diff
@@ -1,16 +1,31 @@
-import { getAllItems } from "@/helpers/db";
+import { getAllItems, createItem } from "@/helpers/db";
 
 export default async function handler(request, response) {
   switch (request.method) {
     case "GET": {
       const items = await getAllItems();
+      if (!items) {
+        response.status(400).json({ message: "Bad request" });
```
Comment (F-Kirchhoff) [inline pages/api/items/index.js:8]: this is not a bad request but an internal server error.

File: pages/items/[id]/index.js [hunk 8]
```diff
@@ -11,11 +11,15 @@ export default function DetailsPage() {
 
   async function handleStatus() {
     const newItem = { ...item, inDiscuss: !item.inDiscuss };
```
Comment (F-Kirchhoff) [inline pages/items/[id]/index.js:13]: I would rename this to "updatedItem" to clarify that you change an already existing item.

**Review by F-Kirchhoff — APPROVED** [review]
I added two suggestions that you should implement at some point. But for now its fine. LGTM 🦊

**Comment by Olga-Mishareva** [comment]
Thank you Felix, I fixed it!

**Pull request merged**